packages/app-mobile/components/screens/Note.tsx

```diff
@@ -139,15 +139,7 @@ class NoteScreenComponent extends BaseScreenComponent {
 			}
 
 			if (this.state.fromShare) {
-				// effectively the same as NAV_BACK but NAV_BACK causes undesired behaviour in this case:
-				// - share to Joplin from some other app
-				// - open Joplin and open any note
-				// - go back -- with NAV_BACK this causes the app to exit rather than just showing notes
-				this.props.dispatch({
-					type: 'NAV_GO',
-					routeName: 'Notes',
-					folderId: this.state.note.parent_id,
-				});
+				ShareExtension.close();
 				return true;
```

Comment:
I don't get what you are doing here. Obviously this is here so that we can go back to the previous screen. If you remove it, the back button is broken. Also why remove `ShareExtension.close` from everywhere? This is what takes care of clearing the share data (which we must do once the share object is processed).

Besides the broken back button, I still have duplications on iOS, which is expected since we no longer clear the data.

I'm going to restore this, and try to get to a point where it work on iOS, and hopefully that won't break Android

Reply:
Ok you removed ShareExtension.close from one place and added it back here. But the logic doesn't make any sense. Why do we only clear it when pressing the back button? Instead that data should cleared as early as possible, once it's been read basically. So I've changed this and added ShareExtension.close earlier